### DISCUSSION TITLE
How to transfer

### DISCUSSION THREAD

**Discussion by goldo1969** [LsHallo/ets2-dedicated-convoy-server#12] [Q&A] [answered]
Hello, how to transfer the 2 files (server packages.dat and server packages.sii) to the docker server from windows. I am new to this system thank you

**Comment by LsHallo** [comment, marked as answer]
Hello,
have you tried not using your own server_packages? I had no problems with the default ones and all DLC I own. I don't think it is important.

If you need to you can start the container like this `[...] -v "C:/local/path:/home/steam/.local/share/Euro Truck Simulator 2" [...]` and then you will see the files appear in your local directory. There you can replace the server_packages.

You can learn more about docker volumes here: https://docs.docker.com/storage/volumes/

**Reply by LsHallo** [reply]
Just to clarify:
You ran `docker run -v "C:/local/path:/home/steam/.local/share/Euro Truck Simulator 2" --network host --restart unless-stopped -e "ETS_SERVER_NAME=My Custom Server" lshallo/ets2-convoy-server` and then you looked at C:\local\path and no files where inside that directory?

**Reply by goldo1969** [reply]
Thank you

**Reply by LsHallo** [reply]
I don't understand. Did you figure it out or do you need more help?

**Reply by goldo1969** [reply]
I would like some help but as I told you, I don't know much about the Docker system.
your image works very well, but I want to be able to add my personal mods.
I don't really know where to put the command you gave me and I don't know how to access the files.
I'm running Promox with Docker...
Thanks
![proxmox](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/87611893/449caafc-cfc6-4614-868c-6c4da4c783d6)
![docker](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/87611893/e4a14787-b4c5-4ea0-a2f2-eb42c701703c)
![path](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/87611893/cb806630-d6bf-4068-87c7-2f728aab18c4)

**Reply by LsHallo** [reply]
Okay I see now. You can remove any try to incorporate the command in portainer.  
The top setting in the volume tab looks about right. You are using a docker volume and not a bind mount though. So you can't access it on a defined local path. Instead it should be in `/var/lib/docker/volumes/[name]/_data`. Alternatively you can set the type (right next to the path) to `bind` and then you can specify a custom path where you can put your config. This would be the preferred way.  
You can also delete the volume with /opt/ets2.


On an unrelated note: I would strongly encourage you to start and learn docker 'the hard way' and not use portainer. I found it more confusing than helpful most of the time and some functionality was straight up missing (this was ~3 years ago now). But if you find it helpful you can stay with it of course.

**Reply by goldo1969** [reply]
I stayed under Portainer and therefore put the volume in bind.
and I now have a legal problem and so I think the config is good.
so I have portainer/docker under promox: should I add the steam user to proxmox and give the steam user rights to this folder?

```
PermissionError: [Errno 13] Permission denied: '/home/steam/.local/share/Euro Truck Simulator 2/server_config.sii'
cp: cannot create regular file '/home/steam/.local/share/Euro Truck Simulator 2/server_packages.sii': Permission denied
cp: cannot create regular file '/home/steam/.local/share/Euro Truck Simulator 2/server_packages.dat': Permission denied
```
![Capture d'écran 2023-10-05 191814](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/87611893/30eea5e8-bfeb-4743-98e6-d95f34bb8287)

**Reply by LsHallo** [reply]
Go to the folder and do `chown -R 1000:1000 .`. It might show a different user in your system but the steam user has user id 1000. That should fix your permission problem.

**Reply by goldo1969** [reply]
as you say, user 1000 does not exist on my system... how do I know the id?
![chown](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/87611893/688895a7-37bf-496d-80d5-5b9054aa57cb)

**Reply by LsHallo** [reply]
The error indicates that you are missing the path component. You can `chown` with the id just fine. You just need to add the path `/var/lib/docker...` after the 1000:1000.  

To get the id of the user open the console in portainer for the container and run the `id` command.

**Reply by goldo1969** [reply]
```
steam@docker:/app$ id
uid=1000(steam) gid=1000(steam) groups=1000(steam)
steam@docker:/app$
```

**Reply by goldo1969** [reply]
```
Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Thu Oct  5 20:25:14 CEST 2023 on pts/0
root@pve:~# cd /var/lib/docker/volumes/ets2
root@pve:/var/lib/docker/volumes/ets2# chown -R 1000:1000 /var/lib/docker/volumes/ets2
root@pve:/var/lib/docker/volumes/ets2#
``` 
the command is apparently good but still the error

```
PermissionError: [Errno 13] Permission denied: '/home/steam/.local/share/Euro Truck Simulator 2/server_config.sii'
cp: cannot create regular file '/home/steam/.local/share/Euro Truck Simulator 2/server_packages.sii': Permission denied
cp: cannot create regular file '/home/steam/.local/share/Euro Truck Simulator 2/server_packages.dat': Permission denied
```

**Reply by goldo1969** [reply]
```
[INFO]: Starting server...
bash: line 1: /app/bin/linux_x64/eurotrucks2_server: No such file or directory
Traceback (most recent call last):
  File "/ets_server_entrypoint.py", line 124, in <module>
[INFO]: Writing config file...
    with open(config_path, "w") as f:
PermissionError: [Errno 13] Permission denied: '/home/steam/.local/share/Euro Truck Simulator 2/server_config.sii'
cp: cannot create regular file '/home/steam/.local/share/Euro Truck Simulator 2/server_packages.sii': Permission denied
cp: cannot create regular file '/home/steam/.local/share/Euro Truck Simulator 2/server_packages.dat': Permission denied
[INFO]: Starting server...
bash: line 1: /app/bin/linux_x64/eurotrucks2_server: No such file or directory
```

**Reply by LsHallo** [reply]
You'll need to log in to your docker virtual machine. The mount will not propagate to the proxmox file system.

There you need to run the chown command. Also maybe change the path to /opt/ets from /var/lib/docker now that you are using a bind mount. Because docker volumes will be created in /var/lib/docler and you don't wanna mess with that.

Rundown:
1. Log in to docker VM
2. Create `/opt/ets2`
3. Run `chown -R 1000:1000 /opt/ets2`
4. Create the container with a bind mount:
`/home/steam/.local/share/Euro Truck Simulator 2` to `/opt/ets2`.
5. The container should start and you should see files in `/opt/ets2`. Replace server_packages as needed.

**Reply by goldo1969** [reply]
Hello, thank you again for taking the time for the explanation.
Is it possible to modify the directory: share/Euro Truck Simulator without spaces so that it is compatible with synology nas, because in the configuration, it is not possible to  spaces.
for example : Euro-Truck-Simulator
on a synology nas the container works very well, but without access to the files because of this folder named with spaces.
Thanks

**Reply by LsHallo** [reply]
> Hello, thank you again for taking the time for the explanation. Is it possible to modify the directory: share/Euro Truck Simulator without spaces so that it is compatible with synology nas, because in the configuration, it is not possible to spaces. for example : Euro-Truck-Simulator on a synology nas the container works very well, but without access to the files because of this folder named with spaces. Thanks

Hello
I don't think I can rename the folder. It gets generated by steam and I have not found a way to change this.
You could install portainer on the synology nas and create the container in portainer. This would allow you to use spaces.
